DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/hardware_reuse/README.md

```diff
@@ -0,0 +1,358 @@
+The following example shows how to use task sequences to
+
+Each task sequence class object represents a specific instantiation of FPGA hardware to perform the operation defined by the 'Task function' parameter.
+Launching tasks via repeated `async()` function calls on the same object tells the compiler to reuse that object's hardware. Thus, you can control the reuse or replication of FPGA hardware by the number of `task_sequence` objects you declare. Since object lifetime is confined to the scope in which the `task_sequence` object is created, carefully declare your object in the scope in which you intend to reuse it.
+
+In the `3_task_sequence` directory, the device code declares the task sequence object once, and invokes it inside the loop, and at the return point.
+
+```c++
+struct VectorOp{
+  int len;
+
+  void operator()() const{
+    constexpr D3Vector coef1 = {0.2, 0.3, 0.4};
+    constexpr D3Vector coef2 = {0.6, 0.7, 0.8};
+
+    D3Vector new_item;
+
+    // Object declarations of a parameterized task_sequence 
+    // class must be local, which means global declarations
+    // and dynamic allocations are not allowed.
+    // Declare the task sequence object outside the for loop 
+    // so that the hardware can be shared at the return point.
+    sycl::ext::intel::experimental::task_sequence<OpSqrt> task_a;
+
+    for (int i = 0; i < len; i++){
+      D3Vector item = InputPipeA::read();
+      task_a.async(item, coef1);
+      new_item[i] = task_a.get();
+    }
+
+    task_a.async(new_item, coef2);
+    OutputPipeZ::write(task_a.get());
+  }
+};
+```
+
+<img src="assets/hardware_reuse_taskseq.svg" />
+
+## Sample Structure
+The 3 different example designs in this sample perform similar operations. You may compare the C++ source files to see the code changes that are necessary to apply hardware reuse.
+
+1. [Naive](1_naive/src/main.cpp) The component directly implements square root of dot product four times and each of them generates the hardware for its own use. 
+2. [Looping](2_loop/src/main.cpp) Square root of dot product with same level of of data path are group together, and invoked in the 'for' loop. The hardware are shared by each invocation in the loop. Another square root of dot product, which is involed at the return point, generates its own hardware.
+3. [Task sequence](3_task_sequence/src/main.cpp) Square root of dot product is invoked with a same task sequence object, both in the loop and at the return point. The hardware of it is shared by each invocation of the task.
+
+## Build the `hardware_reuse` Tutorial
+> **Note**: When working with the command-line interface (CLI), you should configure the oneAPI toolkits using environment variables.
+> Set up your CLI environment by sourcing the `setvars` script located in the root of your oneAPI installation every time you open a new terminal window.
+> This practice ensures that your compiler, libraries, and tools are ready for development.
+>
+> Linux*:
```

Comment:
Out of date information.

Refer to https://www.intel.com/content/www/us/en/docs/oneapi/programming-guide/current/use-the-setvars-and-oneapi-vars-scripts-with-linux.html to get the correct information.